File: fixtures/scale/swift/large/Families/Beaufort/BeaufortDescriptor.swift
struct BeaufortDescriptor: FamilyDescriptor {
    var family: String { "beaufort" }
    var suite: String { "polyalphabetic" }
    var kind: CipherKind { .letter }

    func cipher() -> Cipher {
        return BeaufortCipher()
    }

    func vectors() -> [TestVector] {
        return BeaufortVectors.all()
    }
}

File: fixtures/scale/csharp/large/Families/Beaufort/BeaufortDescriptor.cs
using System.Collections.Generic;
using CipherLab.Abstractions;

namespace CipherLab.Families.Beaufort
{
    public class BeaufortDescriptor : IFamilyDescriptor
    {
        public string Family => "beaufort";
        public string Suite => "polyalphabetic";
        public CipherKind Kind => CipherKind.Letter;

        public ICipher Cipher()
        {
            return new BeaufortCipher();
        }

        public List<TestVector> Vectors()
        {
            return BeaufortVectors.All();
        }
    }
}
